fairscale/optim/oss.py

```diff
@@ -389,17 +379,34 @@ def load_state_dict(self, state_dict: Dict[str, Any]) -> None:
                 from a call to :meth:`state_dict`
         """
 
-        # NOTE: PyTorch 1.5 does not index linearly but with the id(params) at saving time
-        # we work around that here by using the fact that the params are ordered as in the param_groups
-        pytorch15_index_redirect = {k: i for i, k in enumerate(state_dict["state"].keys())}
+        # Update the state, trusting the ordering in param_groups
+        # Apart from the removal of states not owned by this rank, the pytorch logic is kept
+        # (See torch.optim.optimizer)
+        id_map = {
```

Comment:
this is exactly the same logic as pytorch (as commented), which I think is a safer and sane-er (if this word exist ?) way to go